rasa_sdk/forms.py

```diff
@@ -543,9 +543,12 @@ async def _validate_if_required(
             - the form is called after `action_listen`
             - form validation was not cancelled
         """
-        if tracker.latest_action_name == "action_listen" and tracker.active_form.get(
-            "validate", True
-        ):
+        # no active_form means that it is called during activation
+        need_validation = not tracker.active_form or (
+            tracker.latest_action_name == "action_listen"
```

Comment:
let's use the constant `ACTION_LISTEN_NAME` and make sure it's covered by a test

Reply:
the whole `FormAction` uses strings, it'll be weird to use constant only here, and I don't think it makes sense to refactor it, since we're going to deprecate it.

Reply:
> the whole FormAction uses strings, it'll be weird to use constant only her

I strongly disagree. Just because something wasn't done correctly in the past, we shouldn't carry on doing this. 

> and I don't think it makes sense to refactor it, since we're going to deprecate it.

It'll still stay in the code for a while, so why not doing the 10sec code change and be done with it?

Reply:
>  it is user facing code, and constant is way more confusing then string, I'd keep the string here

We already have an existing constant for this in the SDK and we should use it. It might make sense to not use a constant (or delay its introduction) if there is no constant yet and the raw string only appears in very few places. As we already have the constant I see using the constant as a must do.
How is `ACTION_LISTEN_NAME` more confusing than `"action_listen"`?

Reply:
> We already have an existing constant for this in the SDK and we should use it.

this constant is inside `interfaces.py`, quite hard for users to find.

> How is ACTION_LISTEN_NAME more confusing than "action_listen"?

it is super not clear what it is, if you are not familiar with rasa systems, while the string explicitly states that it is some action

Reply:
> this constant is inside interfaces.py, quite hard for users to find.

The import makes it clear where to find + the users shouldn't change this part anyway. 

> it is super not clear what it is, if you are not familiar with rasa systems, while the string explicitly states that it is some action

Both, string and constant, contain the `action_` part so both are clear. 

@degiz @erohmensing @alwx What do you think? I feel this discussion is a bit stuck 😄

Reply:
I'm for using the "ACTION_LISTEN_NAME" here for mentioned reasons:
- I agree that even though we deprecate that it's a tiny change that brings more benefits
- a good habit to not use explicit strings, and this is a good chance to exercise that. If the string has to change at some point of time it's good to have only one constant, not to search for it
- the source file is not being read as a book, users have IDEs or text editors or `grep`

Reply:
in this particular case the source is used as a book because users look at it in GitHub browser and not in IDE